Riot/Modules/Application/AppCoordinator.swift

```diff
@@ -322,15 +322,39 @@ fileprivate class AppNavigator: AppNavigatorProtocol {
         self.appCoordinator.navigate(to: destination)
     }
     
-    func addLoadingActivity() -> Activity {
-        let presenter = ActivityIndicatorToastPresenter(
-            text: VectorL10n.roomParticipantsSecurityLoading,
-            navigationController: appNavigationVC
-        )
-        let request = ActivityRequest(
-            presenter: presenter,
-            dismissal: .manual
-        )
+    func addAppActivity(_ type: AppActivityType) -> Activity {
```

Comment:
Again, `addAppActivity` on the `AppNavigator` level doesn't mean much. What kind of activity is it, all kinds of things can be active on that level?

Reply:
I will also extract this to a separate protocol, however best done in a separate PR, as it creates some issues with dependency injection which will increase the scope of this PR even further.